From 1bf34f176089c11758767d71f7d6d237884b9c1f Mon Sep 17 00:00:00 2001
From: Axel Castaneda Gonzalez <x0055901@ti.com>
Date: Wed, 16 Mar 2011 18:56:43 -0500
Subject: ASoC: HDMI: Shut off audio only when HDMI is active

In order to avoid using HDMI-DSS framework when
HDMI is OFF. HDMI-Audio power is shut off when
HDMI_EVENT_POWERON event is triggered; however, if system
goes to suspend, DAI shutdown was trying to disable
HDMI-Audio resources after DSS clocks were cutoff.
In order to avoid this scenario, HDMI-Audio power
is forced to be released only when HDMI is still active.

This schema is temporal, once HPD events are reliable,
HDMI-Audio resources will be released base on these
notifications.

Change-Id: I73c53259779e283773123514da2ad8d17ba03675
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
Signed-off-by: Axel Castaneda Gonzalez <x0055901@ti.com>
---
 sound/soc/omap/omap-hdmi.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/sound/soc/omap/omap-hdmi.c b/sound/soc/omap/omap-hdmi.c
index 2ba7fd3..e16263e 100644
--- a/sound/soc/omap/omap-hdmi.c
+++ b/sound/soc/omap/omap-hdmi.c
@@ -138,7 +138,9 @@ static void omap_hdmi_dai_shutdown(struct snd_pcm_substream *substream,
 #ifdef CONFIG_HDMI_NO_IP_MODULE
 	hdmi_w1_wrapper_disable(HDMI_WP);
 	hdmi_data.substream = NULL;
-	hdmi_set_audio_power(0);
+
+	if (hdmi_data.active)
+		hdmi_set_audio_power(0);
 #else
 	if (hdmi_audio_core.module_loaded)
 		hdmi_audio_core.wrapper_disable(HDMI_WP);
-- 
1.6.0.4

